### ISSUE TITLE
USWDS - Bug: Range slider missing visible cues for value

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds#6116]
### Describe the bug

The range slider has audible cues that indicate "22 out of 100" for example, but that type of granular detail isn't clear when viewing the range slider. The location of the button doesn't indicate the current value nor does the entire slider bar indicate the capacity (0-100 for example).

### Steps to reproduce the bug

1. Go to https://designsystem.digital.gov/components/range-slider/
2. View the range slider
3. Operate the range slider using JAWS, VO, or NVDA
4. Hear the precise values announced.
5. Note that there are no visible indications that reflect those exact numbers.

### Expected Behavior

We want to give the most accessible example of the component for users to follow. By allowing SR users to hear the precision value of the amount, we are not providing an equal experience for sighted users who instead get a rougher, "ballpark" range experience. Sighted users also are unable to tell the full amount (in the case of our example, it goes from 0-100).

### System setup

JAWS 2024, PC, Chrome

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mlloydbixal** [comment]
I was able to confirm the problem - the screen reader (I used VO) announces the range slider's value but there is no visual representation.

Suggestions for improvements:
- Add a way to represent the range
    - In most cases this would be min and max numbers, however...
    - I wonder how often this range feature gets used with something other than numbers and how we can allow flexibility while still making it more visual (in the ["relative value" example under guidance](https://designsystem.digital.gov/components/range-slider/#range-slider-guidance) I could see someone using icons to interpret less sound / more sound vs number range)
- Add the current selected number or value (maybe above the sliding button or off to the side of the whole slider)

**Comment by amycole501** [comment]
I agree, I think at minimum we should provide guidance around how to correctly represent a value either numerical or otherwise.  Perhaps having another variant for non-numerical values? Or simply mentioning that if you have a non-numerical value then doing so in an accessible way by including alt-text or other instructions.

**Comment by amyleadem** [comment]
## Next steps:
1. Gather accessible implementation examples of the range slider from the landscape
2. Assess if there is a common pattern. 
    1. If there is, gather screenshots and present a proposed visual and markup/aria structure to the accessibility and usability teams for review
    2. If there is not, work with the accessibility and usability teams to determine the preferred component structure
3. Build the component

**Comment by alex-hull** [comment]
Placing recent [range slider findings](https://docs.google.com/document/d/1hF5uEp8eO4UBQtBkksX_G5FASu462-6_qgA62sCgYGc/edit?usp=sharing) for visual representation of values. All explored have some sort of visual value as the user moves up or down the slider. Something we may also need to consider is a text input for the visual value, for those who may not be able to drag the slider button.

**Comment by amyleadem** [comment]
Related note from @amycole501 in #6117:

Following up on this, we received some great feedback from a user who is visually impaired/low vision. We asked them a few questions to test the range slider focus area and these were the responses:

Here is my take on the range slider. I did not attempt to use any keyboard shortcuts with the slider, though I did use Tab and Shift + Tab to see how the focus moved.

1. **Are you able to tell where the focus should be on the slider button? Does it have enough focus?**
   The focus enhancement with ZoomText/Fusion did move to the range slider once I clicked the slider with the mouse.  However, the red rectangle that is my focus enhancement surrounded the entire slider and did not place any particular focus on the handle.
2. **Do you have enough visible cues to operate it?**
   Yes, but the contrast could be better.  I would recommend a thicker line for the slider.  I would also recommend doing a fill for the handle.  That would make it stand out much more against the slider.  I had a little bit of difficulty following it enough to click with the mouse.
3. **Would having a text field to enter a precise numeral into the slider be a good addition?**
   I think this depends on the information being requested.  If a specific value is needed, the text box could be helpful.  However, I tend to think that would mostly clog your page.  If the information requested needs to be that value-specific, a range slider may not be the best element to choose.  If the information requested is of more of a general nature, I think that the increased line thickness and filling in the handle would create enough contrast for a ZoomText/Fusion or Windows Magnifier user to use their vision and the mouse.
4. **Any suggestions for us for improvements to the slider for a low vision user?**
   Just give it as much contrast as possible.  The thin lines that are currently there do make it more difficult.  I looked at it under the same magnification level, but both with and without color enhancements.  Without color enhancements, the lines were too thin for me to really make it out.  Using the invert brightness feature, which turned my background black and made the slider show up white, I was able to see it much easier.

**Comment by dct2023** [comment]
@cathybaptista Can you confirm you are owning this review? Or are you waiting on a review from another team member?

**Comment by dct2023** [comment]
Confirming I talked to Cathy and she shared that Amy L is working on this peer review.

**Comment by annepetersen** [comment]
I believe [this is the PR supporting this work](https://github.com/uswds/uswds/pull/6302) ([Storybook preview link](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cb-range-slider-missing-cues/?path=/story/components-form-inputs-range--range))
